### PR TITLE
Feat : Added category filtering

### PR DESCRIPTION
**Pull request by srejitk**
This feature request adds
1. Category Filtering
2. Bug Fixes

### CODE REVIEW

**Review by shubhamkumar648 — APPROVED** [review]
- perfect use of reducer and context
- code is understandable and looks good
- good use of ternary operator 
- good to merge

File: src/components/CategoryChips/CategoryChips.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { useFilter } from "../../context/Filter/FilterContext";
+import { useUserDetails } from "../../context/User/UserContext";
+import { useVideo } from "../../context/Video/VideoContext";
```
Comment (rdshinde) [inline src/components/CategoryChips/CategoryChips.jsx:4]: Can add index.js for exports in context folder.

File: src/context/Video/Utils.jsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+export const filterByCategory = (state, data) =>
+  state.category === "All"
+    ? data
+    : data?.filter((video) => video.category === state.category);
+
+export const sortByLatest = (state, data) =>
+  state.sortBy === "Latest"
+    ? [...data].sort((a, b) => new Date(b.published) - new Date(a.published))
+    : data;
```
Comment (rdshinde) [inline src/context/Video/Utils.jsx:15]: Good work here.

File: src/pages/History/History.jsx [hunk 3]
```diff
@@ -2,11 +2,34 @@ import React from "react";
...
+        <h1>History</h1>
+        <button
+          // onClick={() => deleteHis(playlistID, userState)}
+          className={`${styles.trash_btn} btn btn_action`}
+        >
+          <span className="material-icons">delete</span>
+        </button>
+        <h4 className="subtitle-1">
+          {/* Total Videos :{" "}
+          {findPlaylist.videos?.length ? findPlaylist.videos?.length : 0} */}
```
Comment (rdshinde) [inline src/pages/History/History.jsx:30]: Remove the comments if not needed.

**Review by rdshinde — APPROVED** [review]
Good work , I liked that you are using `... ? ... :` operator instead of `&&` wherever it is needed.
You can merge.

File: src/components/CategoryChips/CategoryChips.jsx (hunk 1 above)
Comment (srejitk) [reply src/components/CategoryChips/CategoryChips.jsx:4]: Yes, Will do that in the next PR.

File: src/pages/History/History.jsx (hunk 3 above)
Comment (srejitk) [reply src/pages/History/History.jsx:30]: Yes, thanks will do that.

**Pull request merged**